components/value-distribution/src/main/java/org/datacleaner/beans/valuedist/ValueDistributionAnalyzerResultReducer.java

```diff
@@ -0,0 +1,176 @@
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.beans.valuedist;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import org.datacleaner.api.AnalyzerResultReducer;
+import org.datacleaner.api.InputColumn;
+import org.datacleaner.api.Provided;
+import org.datacleaner.result.CompositeValueFrequency;
+import org.datacleaner.result.SingleValueFrequency;
+import org.datacleaner.result.ValueCountList;
+import org.datacleaner.result.ValueCountListImpl;
+import org.datacleaner.result.ValueCountingAnalyzerResult;
+import org.datacleaner.result.ValueFrequency;
+import org.datacleaner.storage.RowAnnotation;
+import org.datacleaner.storage.RowAnnotationFactory;
+import org.datacleaner.storage.RowAnnotationImpl;
+
+/**
+ * A reducer of {@link ValueDistributionAnalyzerResult}s.
+ */
+public class ValueDistributionAnalyzerResultReducer implements AnalyzerResultReducer<ValueDistributionAnalyzerResult> {
+
+    @Inject
+    @Provided
+    RowAnnotationFactory _rowAnnotationFactory;
+
+    @Override
+    public ValueDistributionAnalyzerResult reduce(Collection<? extends ValueDistributionAnalyzerResult> analyzerResults) {
+        final ValueCountList topValues = ValueCountListImpl.emptyList();
+        final Collection<String> uniqueValues = Collections.emptyList();
+        final Map<String, RowAnnotation> annotations = Collections.emptyMap();
+        final InputColumn<?>[] highlightedColumns = null;
+
+        SingleValueDistributionResult reducedResult = new SingleValueDistributionResult(null, topValues, uniqueValues,
+                0, 0, 0, annotations, new RowAnnotationImpl(), _rowAnnotationFactory, highlightedColumns);
+
+        for (ValueDistributionAnalyzerResult partialResult : analyzerResults) {
+            if (partialResult instanceof SingleValueDistributionResult) {
+                final SingleValueDistributionResult singleResult = (SingleValueDistributionResult) partialResult;
+                reducedResult = reduceSingleResult(annotations, highlightedColumns, reducedResult, singleResult);
```

Comment:
I see a pattern here where we reduce only two results at a time. That might cause quite some performance overhead because of moving data out and around between many results. Would be better if the reducer would build one big result in a iterative way based on each result in a for-loop.